File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        unordered_map<string,vector<string>>gzip;
        for(string item:strs){
                string ans=item;
                sort(ans.begin(),ans.end());
                gzip[ans].push_back(item);
        }
        
        vector<vector<string>>ans;
        for(auto item: gzip){
                    ans.push_back(item.second);
        }
        
        return ans;
        
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    
    
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
    
        
            dicti={}
            for item in strs:
                
                word_dic=[0]*27
                for i in item:
                
                    word_dic[ord(i)-97]+=1
                
                if tuple(word_dic) not in dicti:
                    dicti[tuple(word_dic)]=[]
                
                dicti[tuple(word_dic)].append(item)
                
                
            
            ans=[]
            for key,value in dicti.items():
                
                ans.append(value)
                
                
            return ans        
                    
                    